### PR TITLE
Feature/filter donations

### PR DESCRIPTION
**Pull request by stann1**
The default sort order of donations was also changed a bit, because it did not make sense from business perspective.

### CODE REVIEW

File: apps/api/src/donations/donations.service.ts [hunk 1]
```diff
@@ -76,10 +76,21 @@ export class DonationsService {
...
+    if (campaignId) {
+      return await this.prisma.donation.findMany({
+        where: { targetVault: {campaign: {id: campaignId}} },
+        orderBy: [{ createdAt: 'desc' }],
+      })
+    } else {
+      return await this.prisma.donation.findMany({
+        orderBy: [{ createdAt: 'desc' }],
+      })
+    }
```
Comment (igoychev) [inline apps/api/src/donations/donations.service.ts:93]: Just tested: Here there is no need of the if condition, the where clause will return all if the id is undefined.
so leave only this code:
```
return await this.prisma.donation.findMany({
        where: { targetVault: {campaign: {id: campaignId}} },
        orderBy: [{ createdAt: 'desc' }],
      })
```

File: docker-compose.yml [hunk 2]
```diff
@@ -49,7 +49,7 @@ services:
       - pg-db-data:/bitnami/postgresql
 
     ports:
-      - 5432:5432
+      - '${DB_PORT?}:5432'
```
Comment (igoychev) [inline docker-compose.yml:52]: good one too!

**Review by igoychev — CHANGES_REQUESTED** [review]
great improvements!
thanks for adding the .vscode config and showing the APIQuery style. Professional function description too!

File: apps/api/src/donations/donations.service.ts (hunk 1 above)
Comment (stann1) [reply apps/api/src/donations/donations.service.ts:93]: not exactly - if the value is an empty string then there will be an exception. While the routing middleware correctly converts empty string to undefined, this business logic can be reused from anywhere in the code, where such conversion will not be available

Comment (kachar) [reply apps/api/src/donations/donations.service.ts:93]: I think we're good to continue as is with the `campaignId` param

**Pull request merged**